src/modules/endpoint/ResourceURLFinder.js

```diff
@@ -0,0 +1,62 @@
+// @flow
+
+import getExtension from './getExtension'
+import { Parser } from 'htmlparser2'
+import type { FetchMapType } from './sagas/fetchResourceCache'
+import { keyBy, reduce } from 'lodash/collection'
+
+export default class ResourceURLFinder {
+  parser: Parser
+  onAttributeTagFound: (name: string, value: string) => void
+  _foundUrls: Set<string> = new Set<string>()
+
+  init () {
+    this.onAttributeTagFound = (name: string, value: string) => {
+      if (name === 'href' || name === 'src') {
+        if (['png', 'jpg', 'jpeg', 'pdf'].includes(getExtension(value))) {
+          this.foundUrls.add(value)
+        }
+      }
+    }
```

Comment:
Why is this function definition part of init?
also foundUrls should probably be emptied (or initialized) here...

Reply:
You can run findResourceUrls multiple times. Each time the foundUrls is cleared. There is no problem with creating that function in the init() as we only need to create it once.

Reply:
do it :D